packages/clay-badge/src/ClayBadge.soy

```diff
@@ -0,0 +1,110 @@
+{namespace ClayBadge}
+
+/**
+ * This renders a badge.
+ */
+{template .render}
+	{@param? icon: [
+		alignment: string,
+		monospaced: bool,
+		spritemap: string,
+		symbol: string
+	]}
+	{@param? id: string}
+	{@param? label: html|string}
+	{@param? href: string}
+	{@param? shape: string}
+	{@param? size: string}
+	{@param? style: string}
+
+	{let $attributes kind="attributes"}
+		class="badge
+			{sp}badge-{$style ?: 'default'}
+
+			{if $shape == 'pill'}
+				{sp}badge-pill
+			{/if}
+
+			{if $size}
+				{sp}badge-{$size}
+			{/if}
+		"
+
+		{if $id}
+			id="{$id}"
+		{/if}
+
+		{if $href}
+			href="{$href}"
+		{/if}
+	{/let}
+
+	{if $href}
+		<a {$attributes}>
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
+		</a>
+	{else}
+		<span {$attributes}>
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
+		</span>
+	{/if}
+
+{/template}
+
+/**
+ * Renders icon content
+ */
+{template .content}
+	{@param? icon: [
+		alignment: string,
+		monospaced: bool,
+		spritemap: string,
+		symbol: string
+	]}
+	{@param? label: html|string}
+
+	{let $iconAlignment: $icon and $icon.alignment ? $icon.alignment : 'right' /}
+
+	{if $icon and $iconAlignment == 'left'}
```

Comment:
As I can see in Lexicon there's no option for icons in badges to be on the left side of the label. In fact, the only accepted icon is the closing one so maybe we can here remove the icon stuff from the API and use a boolean like `closeable` or something like that and, according to this, render or not the close ClayIcon on the right side of the label. What do you think?

Reply:
Hey @carloslancha @Robert-Frampton,

With the use of `closeable` I would have to add the svg url in the code, that would be a bad practice in my vision, you would have some idea?

```html
{if $closeable}
    <button aria-label="close" class="close" type="button">
        {call ClayIcon.render}
            {param monospaced: false /}
            {param spritemap: '../node_modules/clay/build/images/icons/icons.svg' /}
            {param symbol: 'times' /}
        {/call}
    </button>
{/if}
```

Reply:
Hey @matuzalemsteles I'd say we need to pass the spritemap to the component as usual

Reply:
@carloslancha Is it not better to let `icon`. Pass as before? Since we will use `ClayLink` and `ClayButton`.

Reply:
@matuzalemsteles Since we don't want to provide the feature of use any icon but close one, pass an icon object as paran would be confusing IMHO.